src/analytics.js

```diff
@@ -56,3 +57,28 @@ export function triggerAnalyticsEvent(nodeOrDoc, eventType, opt_vars) {
     analytics.triggerEvent(eventType, opt_vars);
   });
 }
+
+/**
+ * Helper method to create analytics element for specific root.
+ * @param {BaseElement} baseInstance
+ * @param {!JSONType} config
+ */
+export function createAnalyticsForExtension(baseInstance, config) {
+  // Note: Require ampdoc include analytics script
+  analyticsForDocOrNull(baseInstance.element).then(analytics => {
+    if (!analytics) {
+      return;
+    }
+    // Create analytics element;
+    const analyticsElem =
+        baseInstance.win.document.createElement('amp-analytics');
+    const scriptElem = createElementWithAttributes(baseInstance.win.document,
+        'script', {
+          'type': 'application/json',
+        });
+    scriptElem.textContent = JSON.stringify(config);
```

Comment:
Agreed. Then maybe we should first tell if the `AmpAnalytics` element is scoped, then treat the `config` differently. I can address the change in later pr where I add change based on `scoped` value.